src/components/ContactsAdd.js

```diff
@@ -1,39 +1,153 @@
-import { useState } from "react"
+import { useState } from "react";
 import { useNavigate } from "react-router-dom";
 
 function ContactsAdd(props) {
-
   // setContacts and contacts must be passed as props
   // to this component so new contacts can be added to the
   // state
-  const { setContacts, contacts } = props
-
+ 
+  const { setContacts, contacts } = props;
+  const [contact, setContact] = useState({
+    firstName: "",
+    lastName: "",
+    street: "",
+    city: "",
+    Email:"",
+    LinkedIn:"",
+    Twitter:""
+  })
+  
+  
   //TODO: Implement controlled form
   //send POST to json server on form submit
+  const navigate = useNavigate();
+
+  const handleSubmit = (event) => {
+   
+    const submit = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(contact),
+    };
+    fetch("http://localhost:4000/contacts", submit)
+      .then(setContacts([...contacts,contact ]))
+      .then(navigate('/contact'));
+    //  setContacts(contact)
+    event.preventDefault();
+    
+  };
+
+
+  const handleChange = (event) => {
```

Comment:
I suggest looking back at previous lessons to find a way to make the onChange function simpler. Using a bunch of if statements like this makes things hard to read and can become very hard to maintain on larger projects